jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/DefaultControllerEnricherTest.java

```diff
@@ -43,42 +47,68 @@ public class DefaultControllerEnricherTest {
 
     private Map<String, Map<String, Object>> config;
     private EnricherContext context;
+    private Properties properties;
 
     @Before
     public void setUp() throws Exception {
         config = new HashMap<>();
+        properties = new Properties();
         context = JKubeEnricherContext.builder()
             .processorConfig(new ProcessorConfig(null, null, config))
             .log(new KitLogger.SilentLogger())
-            .resources(ResourceConfig.builder().build())
             .image(ImageConfiguration.builder()
                 .name("helloworld")
                 .build(BuildConfiguration.builder()
                     .port("8080")
                     .build()).build())
             .project(JavaProject.builder()
-                .properties(new Properties())
+                .properties(properties)
                 .groupId("group")
                 .artifactId("artifact-id")
                 .build())
             .build();
     }
 
     @Test
-    public void checkReplicaCount() throws Exception {
-        enrichAndAssert(3);
+    public void checkReplicaCount() {
+        givenReplicaCountInEnricherConfig(String.valueOf(3));
+        enrichAndAssert(3, "IfNotPresent");
```

Comment:
For readability purpose, I would split `enrichAndAssert` into 3 different method:
- enrich
- assertReplicatSet
- assertPullPolicy